common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -1346,21 +1346,17 @@ bool mon_wlan_hal_dwpal::process_dwpal_nl_event(struct nl_msg *msg)
             return true;
         }
 
-        // We need to distinguish 1st and rest of dump events (NL80211_CMD_NEW_SCAN_RESULTS)
-        // 1st event is an empty dump result that we should reply with scan dump request.
-        // nlh->nlmsg_seq = 0 only with the 1st dump result.
-        // rest of events are the actual scan dump results that need to be parsed.
-        // unique sequence number is chosen by the nl (nlh->nlmsg_seq != 0) for the rest of events.
+        // Validate dump sequence number is cleared & no scan is in progress
         if (m_nl_seq == 0) {
```

Comment:
I think you should leave the comment "We need to distinguish...".
It helps to understand the flow.
You can add the new comment after it.

You are saying on the new comment that you validate that "no scan is in progress" but I don't see it being validated.

Reply:
IMO you should keep the comment above

Reply:
I re-wrote the comment, with better grammar, and a clearer explanation to the scan dump results process